### PR TITLE
Implemented Event topic model and api, relationship with event 

### LINKED ISSUE #3831: Implement model & API for event types, event topic, event sub-topic
Currently, we have a static array of strings for event types, event topic, event sub-topic. Which is not very efficient and reliable on the longer run especially with a decoupled architecture. We also would like to support customisation by the admin and internalisation of strings too. 

Keeping that in mind, the following are required. 
- Model and API for event type & relationship with event
- Model and API for event topic & relationship with event, event sub-topic
- Model and API for event sub-topic & relationship with event, top
- All of the above with proper migration of existing information

The relationships are as follows:
- One to many relationship b/w event type -> event. (One type has many events)
- One to one relationship b/w event -> event type (One event belongs to one event type)
- One to many relationship b/w event topic -> event sub-topic (One event topic has many sub-topics)
- One to one relationship b/w event sub-topic -> event topic (One event sub-topic belongs to one topic)

Some special requirements for the event API
- The sub-topic, topic, type relationships should always be included in the event response. (Without having to explicitly specify the `include` query param) 

**This is a parent issue. Sub-issues to be created and worked upon.**

### LINKED ISSUE #3841: Implement model & API for event topic
**I'm submitting a ... (check one with "x")**
- [X]  feature request

Parent issue: #3831

- **Model and API for event topic & relationship with event, event sub-topic**

**Relationship:** 
- One to many relationship b/w event topic -> event sub-topic (One event topic has many sub-topics)

**DB:**
- id (Eg. 4) - Unique integer
- name (Eg. Auto, Boat & Air)
- slug (Eg. auto-boat-air) - Unique string (Default: dash-cased version of the name)

### PR DESCRIPTION
**Pull request by magdalenesuo**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `nextgen` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Created Event topic API and Model, relationship with event
(Relationship with subtopic will be continued after making subtopic model and api. Will add in that issue's- #3842  PR)

#### Changes proposed in this pull request:

- One to many relationship b/w event topic -> event. (One topic has many events)
`/event-topics/<int:id>/relationships/events`

- One to one relationship b/w event -> event topic (One event belongs to one event topic)
`/events/<int:id>/relationships/event-topic`

- **endpoints:**
- `/event-topics` [POST and GET] for event type list
- `/event-topics/<int:id>` and `/events/<int:event_id>/event-topic` for event type detail
- `'/event-topics/<int:event_topic_id>/events'` for event list (to get all events with same topic)

Fixes #3841 
Partly fixes #3831

### CODE REVIEW

**Comment by enigmaeth** [comment]
@niranjan94 Why aren't we having event_identifier endpoint for event-topic and event-type? Should it be added?

File: app/api/event_topics.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    data_layer = {'session': db.session,
+                  'model': EventTopic}
+
+
+class EventTopicDetail(ResourceDetail):
+    """
+    Event topic detail by id
+    """
+    def before_get_object(self, view_kwargs):
+        if view_kwargs.get('event_identifier'):
```
Comment (enigmaeth) [inline app/api/event_topics.py]: @magdalenesuo  There is no event_identifier endpoint for event-topic.

File: app/api/event_types.py [hunk 2]
```diff
@@ -52,10 +52,35 @@ class EventTypeDetail(ResourceDetail):
...
+                raise ObjectNotFound({'parameter': 'event_identifier'},
+                                     "Event: {} not found".format(view_kwargs['event_identifier']))
+            else:
+                view_kwargs['event_id'] = event.id
+
+        if view_kwargs.get('event_id'):
+            try:
+                event = self.session.query(Event).filter_by(id=view_kwargs['event_id']).one()
+            except NoResultFound:
+                raise ObjectNotFound({'parameter': 'event_id'},
```
Comment (enigmaeth) [inline app/api/event_types.py]: Add import statements for NoResultFound and ObjectNotFound?

File: app/api/event_types.py [hunk 3]
```diff
@@ -52,10 +52,35 @@ class EventTypeDetail(ResourceDetail):
     """
     Event type detail by id
     """
+    def before_get_object(self, view_kwargs):
+        if view_kwargs.get('event_identifier'):
```
Comment (enigmaeth) [inline app/api/event_types.py]: @magdalenesuo same for event-types here. Please check.

**Comment by niranjan94** [comment]
@enigmaeth yes. Should be added. I missed that. 😓 
@magdalenesuo could you add it ?

**Comment by magdalenesuo** [comment]
@enigmaeth @niranjan94 : yup. I'll add them all in the sub topic pr.

**Pull request merged**